AppleALC/kern_alc.hpp

```diff
@@ -17,8 +17,44 @@ class AlcEnabler {
 public:
 	void init();
 	void deinit();
+	
+	/**
+	 *	Alocate single instance for shared usage and callbacks
+	 */
+	static void createShared();
+	
+	/**
+	 *	Obtain the allocated shared instance
+	 *
+	 *	@return Allocated AlcEnabler instance
+	 */
+	static AlcEnabler* getShared() {
+		return callbackAlc;
+	}
+	
+	/**
+	 *  Hooked IOHDACodecDevice executeVerb
+	 *
+	 *  @param hdaCodecDevice IOHDACodecDevice instance
+	 *  @param nid Node ID
+	 *  @param verb The hda-verb command to send (as defined in hdaverb.h)
+	 *  @param param The parameters for the verb
+	 *
+	 *  @return kIOReturnSuccess on successful execution
+	 */
+	static IOReturn IOHDACodecDevice_executeVerb(void *hdaCodecDevice, uint16_t nid, uint16_t verb, uint16_t param, unsigned int *a4, bool a5);
```

Comment:
`a4` name really needs to be updated, as it is the return value.

`a5` is `waitForSuccess`. When set to `true` for `set format` command (verb 2) it will wait up to 100 times with `IOSleep(1)` inbetween until it succeeds.

Reply:
Thanks for looking into it Vit.